aptos-move/block-executor/src/executor.rs

```diff
@@ -257,14 +257,11 @@ where
                 )
             },
             ExecutionStatus::Abort(err) => {
-                // Abort indicates an unrecoverable VM failure, and should not occur
-                // even due to speculation. Thus, we do not need to finish execution and
-                // can directly return an error.
-                return Err(code_invariant_error(format!(
-                    "FatalVMError from parallel execution {:?} at txn {}",
-                    err, idx_to_execute
-                ))
-                .into());
+                // Abort indicates an unrecoverable VM failure, but currently it seemingly
+                // can occur due to speculative execution (in particular for BlockMetadata txn).
```

Comment:
BlockMetadata txn the first transaction in the block right? 
So,  BlockMetadata txn couldn't abort earlier but could abort now? Did we introduce some code for this project that resulted in a possible abort for BlockMetadata txn?